diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtcopydialog.cpp /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtcopydialog.cpp
--- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtcopydialog.cpp	2009-12-16 12:39:17.000000000 +0200
+++ /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtcopydialog.cpp	2017-01-28 12:35:55.000000000 +0200
@@ -44,19 +44,38 @@
 ** 
 ****************************************************************************/
 
-#include "qtcopydialog.h"
+
+#include <cppdevtk/gui/qt_copy_dialog.hpp>
+
+#if (CPPDEVTK_DISABLE_CPPDEVTK_WARNINGS && CPPDEVTK_COMPILER_MSVC)
+#pragma warning(disable: 4458)	// C4458: declaration of 'var' hides class member
+#endif
+
+#include <cppdevtk/gui/qt_file_copier.hpp>
+#include <cppdevtk/gui/message_box.hpp>
+#include <cppdevtk/base/cassert.hpp>
+#include <cppdevtk/base/unused.hpp>
+#include "ui_qt_copy_dialog.h"
+#include "ui_qt_overwrite_dialog.h"
+#include "ui_qt_other_dialog.h"
+
 #include <QtCore/QDir>
 #include <QtCore/QFile>
 #include <QtCore/QFileInfo>
 #include <QtCore/QTimer>
 #include <QtCore/QDateTime>
-#include <QtGui/QMessageBox>
-#include <QtGui/QStyle>
 #include <QtCore/QEvent>
-#include "qtfilecopier.h"
-#include "ui_qtcopydialog.h"
-#include "ui_qtoverwritedialog.h"
-#include "ui_qtotherdialog.h"
+#include <QtCore/QtGlobal>
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+#include <QtWidgets/QStyle>
+#else
+#include <QtGui/QStyle>
+#endif
+
+
+namespace cppdevtk {
+namespace gui {
+
 
 static QString formatSize(qint64 size)
 {
@@ -186,8 +205,8 @@
 
     void init();
 
-    void error(int id, QtFileCopier::Error error, bool stopped);
-    void stateChanged(QtFileCopier::State state);
+    void error(int id, ::cppdevtk::gui::QtFileCopier::Error error, bool stopped);
+    void stateChanged(::cppdevtk::gui::QtFileCopier::State state);
     void done(bool error);
     void started(int id);
     void dataTransferProgress(int id, qint64 progress);
@@ -415,8 +434,8 @@
 
 void QtCopyDialogPrivate::dataTransferProgress(int id, qint64 progress)
 {
-    Q_ASSERT(lastCurrentId == id);
-    Q_UNUSED(id);
+    CPPDEVTK_ASSERT(lastCurrentId == id);
+    ::cppdevtk::base::SuppressUnusedWarning(id);
     currentProgressTime = startTime.elapsed();
     currentProgress = progress;
     QTimer::singleShot(0, q_ptr, SLOT(showProgress()));
@@ -624,13 +643,21 @@
     window \a flags.
 */
 
-QtCopyDialog::QtCopyDialog(QWidget *parent, Qt::WFlags f)
-    : QDialog(parent, f)
+QtCopyDialog::QtCopyDialog(QWidget *parent,
+#		if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+		Qt::WindowFlags
+#		else
+		Qt::WFlags
+#		endif
+		f)
+    : QDialog(parent, f), WidgetBase()
 {
     d_ptr = new QtCopyDialogPrivate;
     Q_D(QtCopyDialog);
     d->q_ptr = this;
     d->init();
+	
+	SetStyleSheetFromFileCross(":/cppdevtk/gui/res/qss", "qt_copy_dialog");
 
     //setModal(false);
 }
@@ -648,8 +675,14 @@
     \sa setFileCopier()
 */
 
-QtCopyDialog::QtCopyDialog(QtFileCopier *copier, QWidget *parent, Qt::WFlags f)
-    : QDialog(parent, f)
+QtCopyDialog::QtCopyDialog(QtFileCopier *copier, QWidget *parent,
+#		if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+		Qt::WindowFlags
+#		else
+		Qt::WFlags
+#		endif
+		f)
+    : QDialog(parent, f), WidgetBase()
 {
     d_ptr = new QtCopyDialogPrivate;
     Q_D(QtCopyDialog);
@@ -657,6 +690,8 @@
     d->init();
 
     setFileCopier(copier);
+	
+	SetStyleSheetFromFileCross(":/cppdevtk/gui/res/qss", "qt_copy_dialog");
 
     //setModal(false);
 }
@@ -800,11 +835,9 @@
 }
 
 
+}	// namespace gui
+}	// namespace cppdevtk
 
 
-
-
-
-
-#include "qtcopydialog.moc"
-#include "moc_qtcopydialog.cpp"
+#include "qt_copy_dialog.moc"
+#include "moc_qt_copy_dialog.cpp"
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtcopydialog.h /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtcopydialog.h
--- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtcopydialog.h	2009-12-16 12:39:17.000000000 +0200
+++ /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtcopydialog.h	2017-01-28 12:35:54.000000000 +0200
@@ -44,23 +44,51 @@
 ** 
 ****************************************************************************/
 
-#ifndef QTCOPYDIALOG_H
-#define QTCOPYDIALOG_H
 
+#ifndef CPPDEVTK_GUI_QT_COPY_DIALOG_HPP_INCLUDED_
+#define CPPDEVTK_GUI_QT_COPY_DIALOG_HPP_INCLUDED_
+
+
+#include "config.hpp"
+#include "widget_base.hpp"
+#include "qt_file_copier.hpp"
+
+#include <QtCore/Qt>
+#include <QtCore/QtGlobal>
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+#include <QtWidgets/QDialog>
+#else
 #include <QtGui/QDialog>
-#include "qtfilecopier.h"
+#endif
+
+
+namespace cppdevtk {
+namespace gui {
+
 
 class QtCopyDialogPrivate;
 
-class QT_QTCOPYDIALOG_EXPORT QtCopyDialog : public QDialog
+
+class CPPDEVTK_GUI_API QtCopyDialog : public QDialog, public WidgetBase
 {
     Q_OBJECT
     Q_PROPERTY(int minimumDuration READ minimumDuration WRITE setMinimumDuration)
     Q_PROPERTY(bool autoClose READ autoClose WRITE setAutoClose)
 public:
-
-    QtCopyDialog(QWidget *parent = 0, Qt::WFlags f = 0);
-    QtCopyDialog(QtFileCopier *copier, QWidget *parent = 0, Qt::WFlags f = 0);
+    QtCopyDialog(QWidget *parent = 0,
+#		if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+		Qt::WindowFlags
+#		else
+		Qt::WFlags
+#		endif
+		f = 0);
+    QtCopyDialog(QtFileCopier *copier, QWidget *parent = 0,
+#		if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+		Qt::WindowFlags
+#		else
+		Qt::WFlags
+#		endif
+		f = 0);
     ~QtCopyDialog();
 
     void setFileCopier(QtFileCopier *copier);
@@ -82,8 +110,8 @@
     Q_DECLARE_PRIVATE(QtCopyDialog)
     Q_DISABLE_COPY(QtCopyDialog)
 
-    Q_PRIVATE_SLOT(d_func(), void error(int, QtFileCopier::Error, bool))
-    Q_PRIVATE_SLOT(d_func(), void stateChanged(QtFileCopier::State))
+    Q_PRIVATE_SLOT(d_func(), void error(int, ::cppdevtk::gui::QtFileCopier::Error, bool))
+    Q_PRIVATE_SLOT(d_func(), void stateChanged(::cppdevtk::gui::QtFileCopier::State))
     Q_PRIVATE_SLOT(d_func(), void done(bool))
     Q_PRIVATE_SLOT(d_func(), void started(int))
     Q_PRIVATE_SLOT(d_func(), void dataTransferProgress(int, qint64))
@@ -94,4 +122,9 @@
 
 };
 
-#endif
+
+}	// namespace gui
+}	// namespace cppdevtk
+
+
+#endif	// CPPDEVTK_GUI_QT_COPY_DIALOG_HPP_INCLUDED_
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtcopydialog.ui /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtcopydialog.ui
--- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtcopydialog.ui	2009-12-16 12:39:06.000000000 +0200
+++ /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtcopydialog.ui	2017-03-02 18:37:20.000000000 +0200
@@ -1,106 +1,114 @@
-<ui version="4.0" >
- <author></author>
- <comment></comment>
- <exportmacro></exportmacro>
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
  <class>QtCopyDialog</class>
- <widget class="QDialog" name="QtCopyDialog" >
-  <property name="geometry" >
+ <widget class="QDialog" name="QtCopyDialog">
+  <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>428</width>
-    <height>147</height>
+    <width>419</width>
+    <height>244</height>
    </rect>
   </property>
-  <property name="maximumSize" >
+  <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>16777215</height>
    </size>
   </property>
-  <property name="windowTitle" >
+  <property name="windowTitle">
    <string>24% of 100MB (Copying...)</string>
   </property>
-  <layout class="QVBoxLayout" >
-   <property name="margin" >
+  <layout class="QVBoxLayout">
+   <property name="spacing">
+    <number>6</number>
+   </property>
+   <property name="leftMargin">
     <number>8</number>
    </property>
-   <property name="spacing" >
-    <number>6</number>
+   <property name="topMargin">
+    <number>8</number>
+   </property>
+   <property name="rightMargin">
+    <number>8</number>
+   </property>
+   <property name="bottomMargin">
+    <number>8</number>
    </property>
    <item>
-    <layout class="QGridLayout" >
-     <property name="margin" >
+    <layout class="QGridLayout">
+     <property name="leftMargin">
       <number>0</number>
      </property>
-     <property name="spacing" >
+     <property name="topMargin">
+      <number>0</number>
+     </property>
+     <property name="rightMargin">
+      <number>0</number>
+     </property>
+     <property name="bottomMargin">
+      <number>0</number>
+     </property>
+     <property name="spacing">
       <number>6</number>
      </property>
-     <item row="1" column="0" >
-      <widget class="QLabel" name="destLabel" >
-       <property name="sizePolicy" >
-        <sizepolicy>
-         <hsizetype>0</hsizetype>
-         <vsizetype>5</vsizetype>
+     <item row="1" column="0">
+      <widget class="QLabel" name="destLabel">
+       <property name="sizePolicy">
+        <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
-       <property name="text" >
+       <property name="text">
         <string>Destination</string>
        </property>
       </widget>
      </item>
-     <item row="0" column="1" >
-      <widget class="QLineEdit" name="sourceLineEdit" >
-       <property name="sizePolicy" >
-        <sizepolicy>
-         <hsizetype>7</hsizetype>
-         <vsizetype>0</vsizetype>
+     <item row="0" column="1">
+      <widget class="QLineEdit" name="sourceLineEdit">
+       <property name="sizePolicy">
+        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
-       <property name="focusPolicy" >
+       <property name="focusPolicy">
         <enum>Qt::NoFocus</enum>
        </property>
-       <property name="readOnly" >
+       <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
-     <item row="0" column="0" >
-      <widget class="QLabel" name="sourceLabel" >
-       <property name="sizePolicy" >
-        <sizepolicy>
-         <hsizetype>0</hsizetype>
-         <vsizetype>5</vsizetype>
+     <item row="0" column="0">
+      <widget class="QLabel" name="sourceLabel">
+       <property name="sizePolicy">
+        <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
-       <property name="text" >
+       <property name="text">
         <string>Source</string>
        </property>
       </widget>
      </item>
-     <item row="1" column="1" >
-      <widget class="QLineEdit" name="destLineEdit" >
-       <property name="sizePolicy" >
-        <sizepolicy>
-         <hsizetype>7</hsizetype>
-         <vsizetype>0</vsizetype>
+     <item row="1" column="1">
+      <widget class="QLineEdit" name="destLineEdit">
+       <property name="sizePolicy">
+        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
-       <property name="focusPolicy" >
+       <property name="focusPolicy">
         <enum>Qt::NoFocus</enum>
        </property>
-       <property name="text" >
+       <property name="text">
         <string/>
        </property>
-       <property name="readOnly" >
+       <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
@@ -108,57 +116,66 @@
     </layout>
    </item>
    <item>
-    <widget class="QProgressBar" name="currentProgressBar" >
-     <property name="value" >
+    <widget class="QProgressBar" name="currentProgressBar">
+     <property name="value">
       <number>24</number>
      </property>
     </widget>
    </item>
    <item>
-    <widget class="QProgressBar" name="totalProgressBar" >
-     <property name="value" >
+    <widget class="QProgressBar" name="totalProgressBar">
+     <property name="value">
       <number>24</number>
      </property>
     </widget>
    </item>
    <item>
-    <layout class="QGridLayout" >
-     <property name="margin" >
+    <layout class="QGridLayout">
+     <property name="leftMargin">
       <number>0</number>
      </property>
-     <property name="spacing" >
+     <property name="topMargin">
+      <number>0</number>
+     </property>
+     <property name="rightMargin">
+      <number>0</number>
+     </property>
+     <property name="bottomMargin">
+      <number>0</number>
+     </property>
+     <property name="spacing">
       <number>6</number>
      </property>
-     <item row="0" column="0" >
-      <widget class="QLabel" name="completedLabel" >
-       <property name="text" >
+     <item row="0" column="0">
+      <widget class="QLabel" name="completedLabel">
+       <property name="text">
         <string>10 MB of 100 MB completed</string>
        </property>
       </widget>
      </item>
-     <item row="0" column="1" >
-      <widget class="QLabel" name="filesLabel" >
-       <property name="text" >
+     <item row="0" column="1">
+      <widget class="QLabel" name="filesLabel">
+       <property name="text">
         <string>2 / 5 files</string>
        </property>
-       <property name="alignment" >
+       <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
-     <item row="1" column="0" >
-      <widget class="QLabel" name="remainingLabel" >
-       <property name="text" >
+     <item row="1" column="0">
+      <widget class="QLabel" name="remainingLabel">
+       <property name="text">
         <string>5 MB/s (00:00:15 remaining)</string>
        </property>
       </widget>
      </item>
-     <item row="1" column="1" >
-      <widget class="QLabel" name="dirsLabel" >
-       <property name="text" >
+     <item row="1" column="1">
+      <widget class="QLabel" name="dirsLabel">
+       <property name="text">
         <string>0 / 1 dirs</string>
        </property>
-       <property name="alignment" >
+       <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
@@ -166,19 +183,28 @@
     </layout>
    </item>
    <item>
-    <layout class="QHBoxLayout" >
-     <property name="margin" >
+    <layout class="QHBoxLayout">
+     <property name="spacing">
+      <number>6</number>
+     </property>
+     <property name="leftMargin">
       <number>0</number>
      </property>
-     <property name="spacing" >
-      <number>6</number>
+     <property name="topMargin">
+      <number>0</number>
+     </property>
+     <property name="rightMargin">
+      <number>0</number>
+     </property>
+     <property name="bottomMargin">
+      <number>0</number>
      </property>
      <item>
       <spacer>
-       <property name="orientation" >
+       <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
-       <property name="sizeHint" >
+       <property name="sizeHint" stdset="0">
         <size>
          <width>221</width>
          <height>31</height>
@@ -187,18 +213,18 @@
       </spacer>
      </item>
      <item>
-      <widget class="QPushButton" name="cancelButton" >
-       <property name="enabled" >
+      <widget class="QPushButton" name="cancelButton">
+       <property name="enabled">
         <bool>false</bool>
        </property>
-       <property name="text" >
+       <property name="text">
         <string>Cancel</string>
        </property>
       </widget>
      </item>
      <item>
-      <widget class="QPushButton" name="closeButton" >
-       <property name="text" >
+      <widget class="QPushButton" name="closeButton">
+       <property name="text">
         <string>Close</string>
        </property>
       </widget>
@@ -207,7 +233,6 @@
    </item>
   </layout>
  </widget>
- <pixmapfunction></pixmapfunction>
  <resources/>
  <connections>
   <connection>
@@ -216,11 +241,11 @@
    <receiver>QtCopyDialog</receiver>
    <slot>reject()</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>369</x>
      <y>253</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>179</x>
      <y>282</y>
     </hint>
@@ -232,11 +257,11 @@
    <receiver>QtCopyDialog</receiver>
    <slot>reject()</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>389</x>
      <y>177</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>220</x>
      <y>101</y>
     </hint>
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtfilecopier.cpp /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtfilecopier.cpp
--- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtfilecopier.cpp	2009-12-16 12:39:17.000000000 +0200
+++ /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtfilecopier.cpp	2017-01-28 12:35:55.000000000 +0200
@@ -44,7 +44,19 @@
 ** 
 ****************************************************************************/
 
-#include "qtfilecopier.h"
+
+#include <cppdevtk/gui/qt_file_copier.hpp>
+
+#if (CPPDEVTK_DISABLE_CPPDEVTK_WARNINGS && CPPDEVTK_COMPILER_MSVC)
+#pragma warning(disable: 4456)	// C4456: declaration of 'var' hides previous local declaration
+#pragma warning(disable: 4457)	// C4457: declaration of 'var' hides function parameter
+#pragma warning(disable: 4458)	// C4458: declaration of 'var' hides class member
+#endif
+
+#include <cppdevtk/base/cassert.hpp>
+#include <cppdevtk/base/unused.hpp>
+#include <cppdevtk/base/logger.hpp>
+
 #include <QtCore/QThread>
 #include <QtCore/QMutex>
 #include <QtCore/QWaitCondition>
@@ -57,6 +69,12 @@
 #include <QtCore/QDir>
 #include <QtCore/QTimer>
 #include <QtCore/QMetaType>
+#include <QtCore/QtGlobal>
+
+
+namespace cppdevtk {
+namespace gui {
+
 
 struct CopyRequest {
     CopyRequest() {
@@ -152,7 +170,7 @@
 
     void progress();
 signals:
-    void error(int id, QtFileCopier::Error error, bool stopped);
+    void error(int id, ::cppdevtk::gui::QtFileCopier::Error error, bool stopped);
     void started(int id);
     void dataTransferProgress(int id, qint64 progress);
     void finished(int id, bool error);
@@ -177,7 +195,7 @@
     bool overwriteAllRequest;
     bool cancelRequest;
     int currentId;
-#if QT_VERSION >= 0x040400
+#if (QT_VERSION >= QT_VERSION_CHECK(4, 4, 0))
     QAtomicInt progressRequest;
 #else
     QAtomic progressRequest;
@@ -196,7 +214,9 @@
       currentId(-1),
       autoReset(true)
 {
-    qRegisterMetaType<QtFileCopier::Error>("QtFileCopier::Error");
+	// NOTE: do not global qualify because moc will generate bad code
+    qRegisterMetaType<cppdevtk::gui::QtFileCopier::Error>("cppdevtk::gui::QtFileCopier::Error");
+	
     connect(this, SIGNAL(error(int, QtFileCopier::Error, bool)),
                 copier, SLOT(copyError(int, QtFileCopier::Error, bool)));
     connect(this, SIGNAL(started(int)),
@@ -537,7 +557,7 @@
         if (r.move) {
             QFileInfo fis(r.source);
             QDir dir = fis.dir();
-            if (!(r.copyFlags & QtFileCopier::FollowLinks) || !r.dir && !fis.isSymLink()) {
+            if (!(r.copyFlags & QtFileCopier::FollowLinks) || (!r.dir && !fis.isSymLink())) {
                 if (dir.rename(fis.fileName(), r.dest)) {
                     QFileInfo fid(r.dest);
                     if (r.dir)
@@ -880,7 +900,7 @@
     QtFileCopierPrivate();
 
     void setState(QtFileCopier::State s);
-    void copyError(int id, QtFileCopier::Error error, bool stopped);
+    void copyError(int id, ::cppdevtk::gui::QtFileCopier::Error error, bool stopped);
     void copyStarted(int id);
     void copyFinished(int id, bool err);
     void copyCanceled();
@@ -938,7 +958,7 @@
     state = s;
 }
 
-void QtFileCopierPrivate::copyError(int id, QtFileCopier::Error error, bool stopped)
+void QtFileCopierPrivate::copyError(int id, ::cppdevtk::gui::QtFileCopier::Error error, bool stopped)
 {
     Q_Q(QtFileCopier);
     if (stopped == true)
@@ -958,8 +978,8 @@
 {
     Q_Q(QtFileCopier);
     int pop = currentStack.pop();
-    Q_ASSERT(pop == id);
-    Q_UNUSED(pop);
+    CPPDEVTK_ASSERT(pop == id);
+    ::cppdevtk::base::SuppressUnusedWarning(pop);
     emit q->finished(id, err);
     if (err) {
         error = err;
@@ -1665,11 +1685,11 @@
                 CopyFlags flags)
 {
     if (flags & QtFileCopier::MakeLinks) {
-        qWarning("QtFileCopier: cannot move with MakeLinks option specified, option cleared.");
+        CPPDEVTK_LOG_WARN("cannot move with MakeLinks option specified, option cleared.");
         flags &= ~QtFileCopier::MakeLinks;
     }
     if (flags & QtFileCopier::FollowLinks) {
-        qWarning("QtFileCopier: cannot move with FollowLinks option specified, option cleared.");
+        CPPDEVTK_LOG_WARN("cannot move with FollowLinks option specified, option cleared.");
         flags &= ~QtFileCopier::FollowLinks;
     }
     QFileInfo fis(sourceFile);
@@ -1707,11 +1727,11 @@
                 const QString &destinationDir, CopyFlags flags)
 {
     if (flags & QtFileCopier::MakeLinks) {
-        qWarning("QtFileCopier: cannot move with MakeLinks option specified, option cleared.");
+        CPPDEVTK_LOG_WARN("cannot move with MakeLinks option specified, option cleared.");
         flags &= ~QtFileCopier::MakeLinks;
     }
     if (flags & QtFileCopier::FollowLinks) {
-        qWarning("QtFileCopier: cannot move with FollowLinks option specified, option cleared.");
+        CPPDEVTK_LOG_WARN("cannot move with FollowLinks option specified, option cleared.");
         flags &= ~QtFileCopier::FollowLinks;
     }
     return d_ptr->copyFiles(sourceFiles, destinationDir, flags, true);
@@ -1746,11 +1766,11 @@
                 const QString &destinationDir, CopyFlags flags)
 {
     if (flags & QtFileCopier::MakeLinks) {
-        qWarning("QtFileCopier: cannot move with MakeLinks option specified, option cleared.");
+        CPPDEVTK_LOG_WARN("cannot move with MakeLinks option specified, option cleared.");
         flags &= ~QtFileCopier::MakeLinks;
     }
     if (flags & QtFileCopier::FollowLinks) {
-        qWarning("QtFileCopier: cannot move with FollowLinks option specified, option cleared.");
+        CPPDEVTK_LOG_WARN("cannot move with FollowLinks option specified, option cleared.");
         flags &= ~QtFileCopier::FollowLinks;
     }
     return d_ptr->copyDirectory(sourceDir, destinationDir, flags, true);
@@ -2098,5 +2118,10 @@
     d->progressTimer->setInterval(ms);
 }
 
-#include "qtfilecopier.moc"
-#include "moc_qtfilecopier.cpp"
+
+}	// namespace gui
+}	// namespace cppdevtk
+
+
+#include "qt_file_copier.moc"
+#include "moc_qt_file_copier.cpp"
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtfilecopier.h /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtfilecopier.h
--- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtfilecopier.h	2009-12-16 12:39:17.000000000 +0200
+++ /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtfilecopier.h	2017-01-28 12:35:54.000000000 +0200
@@ -44,30 +44,23 @@
 ** 
 ****************************************************************************/
 
-#ifndef QTFILECOPIER_H
-#define QTFILECOPIER_H
+
+#ifndef CPPDEVTK_GUI_QT_FILE_COPIER_HPP_INCLUDED_
+#define CPPDEVTK_GUI_QT_FILE_COPIER_HPP_INCLUDED_
+
+
+#include "config.hpp"
 
 #include <QtCore/QObject>
 
-#if defined(Q_WS_WIN)
-#  if !defined(QT_QTCOPYDIALOG_EXPORT) && !defined(QT_QTCOPYDIALOG_IMPORT)
-#    define QT_QTCOPYDIALOG_EXPORT
-#  elif defined(QT_QTCOPYDIALOG_IMPORT)
-#    if defined(QT_QTCOPYDIALOG_EXPORT)
-#      undef QT_QTCOPYDIALOG_EXPORT
-#    endif
-#    define QT_QTCOPYDIALOG_EXPORT __declspec(dllimport)
-#  elif defined(QT_QTCOPYDIALOG_EXPORT)
-#    undef QT_QTCOPYDIALOG_EXPORT
-#    define QT_QTCOPYDIALOG_EXPORT __declspec(dllexport)
-#  endif
-#else
-#  define QT_QTCOPYDIALOG_EXPORT
-#endif
+
+namespace cppdevtk {
+namespace gui {
+
 
 class QtFileCopierPrivate;
 
-class QT_QTCOPYDIALOG_EXPORT QtFileCopier : public QObject
+class CPPDEVTK_GUI_API QtFileCopier : public QObject
 {
     Q_OBJECT
     Q_PROPERTY(int progressInterval READ progressInterval WRITE setProgressInterval)
@@ -155,9 +148,9 @@
     void resetOverwrite();
 
 Q_SIGNALS:
-    void error(int id, QtFileCopier::Error error, bool stopped);
+    void error(int id, ::cppdevtk::gui::QtFileCopier::Error error, bool stopped);
 
-    void stateChanged(QtFileCopier::State state);
+    void stateChanged(::cppdevtk::gui::QtFileCopier::State state);
 
     void done(bool error);
     void started(int id);
@@ -174,10 +167,15 @@
     Q_PRIVATE_SLOT(d_func(), void copyStarted(int))
     Q_PRIVATE_SLOT(d_func(), void copyFinished(int, bool))
     Q_PRIVATE_SLOT(d_func(), void copyCanceled())
-    Q_PRIVATE_SLOT(d_func(), void copyError(int, QtFileCopier::Error, bool))
+    Q_PRIVATE_SLOT(d_func(), void copyError(int, ::cppdevtk::gui::QtFileCopier::Error, bool))
     Q_PRIVATE_SLOT(d_func(), void progressRequest())
 };
 
 Q_DECLARE_OPERATORS_FOR_FLAGS(QtFileCopier::CopyFlags)
 
-#endif
+
+}	// namespace gui
+}	// namespace cppdevtk
+
+
+#endif	// CPPDEVTK_GUI_QT_FILE_COPIER_HPP_INCLUDED_
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtotherdialog.ui /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtotherdialog.ui
--- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtotherdialog.ui	2009-12-16 12:39:06.000000000 +0200
+++ /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtotherdialog.ui	2017-03-02 18:37:20.000000000 +0200
@@ -1,91 +1,103 @@
-<ui version="4.0" >
- <author></author>
- <comment></comment>
- <exportmacro></exportmacro>
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
  <class>QtOtherDialog</class>
- <widget class="QDialog" name="QtOtherDialog" >
-  <property name="geometry" >
+ <widget class="QDialog" name="QtOtherDialog">
+  <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>267</width>
-    <height>100</height>
+    <width>416</width>
+    <height>154</height>
    </rect>
   </property>
-  <property name="windowTitle" >
+  <property name="windowTitle">
    <string>File Already Exists</string>
   </property>
-  <layout class="QVBoxLayout" >
-   <property name="margin" >
+  <layout class="QVBoxLayout">
+   <property name="spacing">
+    <number>6</number>
+   </property>
+   <property name="leftMargin">
     <number>8</number>
    </property>
-   <property name="spacing" >
-    <number>6</number>
+   <property name="topMargin">
+    <number>8</number>
+   </property>
+   <property name="rightMargin">
+    <number>8</number>
+   </property>
+   <property name="bottomMargin">
+    <number>8</number>
    </property>
    <item>
-    <layout class="QGridLayout" >
-     <property name="margin" >
+    <layout class="QGridLayout">
+     <property name="leftMargin">
       <number>0</number>
      </property>
-     <property name="spacing" >
+     <property name="topMargin">
+      <number>0</number>
+     </property>
+     <property name="rightMargin">
+      <number>0</number>
+     </property>
+     <property name="bottomMargin">
+      <number>0</number>
+     </property>
+     <property name="spacing">
       <number>6</number>
      </property>
-     <item rowspan="2" row="0" column="0" >
-      <widget class="QLabel" name="iconLabel" >
-       <property name="text" >
+     <item row="0" column="0" rowspan="2">
+      <widget class="QLabel" name="iconLabel">
+       <property name="text">
         <string/>
        </property>
       </widget>
      </item>
-     <item row="1" column="2" >
-      <widget class="QLineEdit" name="destLineEdit" >
-       <property name="sizePolicy" >
-        <sizepolicy>
-         <hsizetype>7</hsizetype>
-         <vsizetype>0</vsizetype>
+     <item row="1" column="2">
+      <widget class="QLineEdit" name="destLineEdit">
+       <property name="sizePolicy">
+        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
-       <property name="focusPolicy" >
+       <property name="focusPolicy">
         <enum>Qt::NoFocus</enum>
        </property>
-       <property name="text" >
+       <property name="text">
         <string/>
        </property>
-       <property name="readOnly" >
+       <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
-     <item row="0" column="1" >
-      <widget class="QLabel" name="label" >
-       <property name="text" >
+     <item row="0" column="1">
+      <widget class="QLabel" name="label">
+       <property name="text">
         <string>Source</string>
        </property>
       </widget>
      </item>
-     <item row="0" column="2" >
-      <widget class="QLineEdit" name="sourceLineEdit" >
-       <property name="sizePolicy" >
-        <sizepolicy>
-         <hsizetype>7</hsizetype>
-         <vsizetype>0</vsizetype>
+     <item row="0" column="2">
+      <widget class="QLineEdit" name="sourceLineEdit">
+       <property name="sizePolicy">
+        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
-       <property name="focusPolicy" >
+       <property name="focusPolicy">
         <enum>Qt::NoFocus</enum>
        </property>
-       <property name="readOnly" >
+       <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
-     <item row="1" column="1" >
-      <widget class="QLabel" name="label_3" >
-       <property name="text" >
+     <item row="1" column="1">
+      <widget class="QLabel" name="label_3">
+       <property name="text">
         <string>Destination</string>
        </property>
       </widget>
@@ -93,29 +105,38 @@
     </layout>
    </item>
    <item>
-    <widget class="QLabel" name="messageLabel" >
-     <property name="text" >
+    <widget class="QLabel" name="messageLabel">
+     <property name="text">
       <string>Destination file already exists.</string>
      </property>
-     <property name="alignment" >
+     <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
-    <layout class="QHBoxLayout" >
-     <property name="margin" >
+    <layout class="QHBoxLayout">
+     <property name="spacing">
+      <number>6</number>
+     </property>
+     <property name="leftMargin">
       <number>0</number>
      </property>
-     <property name="spacing" >
-      <number>6</number>
+     <property name="topMargin">
+      <number>0</number>
+     </property>
+     <property name="rightMargin">
+      <number>0</number>
+     </property>
+     <property name="bottomMargin">
+      <number>0</number>
      </property>
      <item>
       <spacer>
-       <property name="orientation" >
+       <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
-       <property name="sizeHint" >
+       <property name="sizeHint" stdset="0">
         <size>
          <width>16</width>
          <height>31</height>
@@ -124,39 +145,39 @@
       </spacer>
      </item>
      <item>
-      <widget class="QPushButton" name="retryButton" >
-       <property name="text" >
+      <widget class="QPushButton" name="retryButton">
+       <property name="text">
         <string>Retry</string>
        </property>
       </widget>
      </item>
      <item>
-      <widget class="QPushButton" name="skipButton" >
-       <property name="text" >
+      <widget class="QPushButton" name="skipButton">
+       <property name="text">
         <string>Skip</string>
        </property>
       </widget>
      </item>
      <item>
-      <widget class="QPushButton" name="skipAllButton" >
-       <property name="text" >
+      <widget class="QPushButton" name="skipAllButton">
+       <property name="text">
         <string>Skip All</string>
        </property>
       </widget>
      </item>
      <item>
-      <widget class="QPushButton" name="cancelButton" >
-       <property name="text" >
+      <widget class="QPushButton" name="cancelButton">
+       <property name="text">
         <string>Cancel</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer>
-       <property name="orientation" >
+       <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
-       <property name="sizeHint" >
+       <property name="sizeHint" stdset="0">
         <size>
          <width>16</width>
          <height>20</height>
@@ -167,8 +188,8 @@
     </layout>
    </item>
   </layout>
-  <widget class="QWidget" name="" >
-   <property name="geometry" >
+  <widget class="QWidget" name="widget1" native="true">
+   <property name="geometry">
     <rect>
      <x>9</x>
      <y>83</y>
@@ -177,8 +198,8 @@
     </rect>
    </property>
   </widget>
-  <widget class="QWidget" name="" >
-   <property name="geometry" >
+  <widget class="QWidget" name="widget2" native="true">
+   <property name="geometry">
     <rect>
      <x>9</x>
      <y>154</y>
@@ -187,8 +208,8 @@
     </rect>
    </property>
   </widget>
-  <widget class="QWidget" name="" >
-   <property name="geometry" >
+  <widget class="QWidget" name="widget3" native="true">
+   <property name="geometry">
     <rect>
      <x>9</x>
      <y>9</y>
@@ -197,8 +218,8 @@
     </rect>
    </property>
   </widget>
-  <widget class="QWidget" name="" >
-   <property name="geometry" >
+  <widget class="QWidget" name="widget4" native="true">
+   <property name="geometry">
     <rect>
      <x>16</x>
      <y>10</y>
@@ -207,8 +228,8 @@
     </rect>
    </property>
   </widget>
-  <widget class="QWidget" name="" >
-   <property name="geometry" >
+  <widget class="QWidget" name="widget5" native="true">
+   <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
@@ -218,7 +239,6 @@
    </property>
   </widget>
  </widget>
- <pixmapfunction></pixmapfunction>
  <resources/>
  <connections/>
 </ui>
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtoverwritedialog.ui /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtoverwritedialog.ui
--- /home/canita/work/patches/qtcopydialog-2.2_1-opensource-orig/src/qtoverwritedialog.ui	2009-12-16 12:39:06.000000000 +0200
+++ /home/canita/work/patches/qtcopydialog-2.2_1-opensource/src/qtoverwritedialog.ui	2017-03-02 18:37:20.000000000 +0200
@@ -1,113 +1,134 @@
-<ui version="4.0" >
- <author></author>
- <comment></comment>
- <exportmacro></exportmacro>
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
  <class>QtOverwriteDialog</class>
- <widget class="QDialog" name="QtOverwriteDialog" >
-  <property name="geometry" >
+ <widget class="QDialog" name="QtOverwriteDialog">
+  <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>305</width>
-    <height>83</height>
+    <width>573</width>
+    <height>156</height>
    </rect>
   </property>
-  <property name="windowTitle" >
+  <property name="windowTitle">
    <string>Copy Warning</string>
   </property>
-  <layout class="QVBoxLayout" >
-   <property name="margin" >
+  <layout class="QVBoxLayout">
+   <property name="spacing">
+    <number>6</number>
+   </property>
+   <property name="leftMargin">
     <number>8</number>
    </property>
-   <property name="spacing" >
-    <number>6</number>
+   <property name="topMargin">
+    <number>8</number>
+   </property>
+   <property name="rightMargin">
+    <number>8</number>
+   </property>
+   <property name="bottomMargin">
+    <number>8</number>
    </property>
    <item>
-    <layout class="QHBoxLayout" >
-     <property name="margin" >
+    <layout class="QHBoxLayout">
+     <property name="spacing">
+      <number>6</number>
+     </property>
+     <property name="leftMargin">
       <number>0</number>
      </property>
-     <property name="spacing" >
-      <number>6</number>
+     <property name="topMargin">
+      <number>0</number>
+     </property>
+     <property name="rightMargin">
+      <number>0</number>
+     </property>
+     <property name="bottomMargin">
+      <number>0</number>
      </property>
      <item>
-      <widget class="QLabel" name="iconLabel" >
-       <property name="text" >
+      <widget class="QLabel" name="iconLabel">
+       <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
-      <layout class="QGridLayout" >
-       <property name="margin" >
+      <layout class="QGridLayout">
+       <property name="leftMargin">
         <number>0</number>
        </property>
-       <property name="spacing" >
+       <property name="topMargin">
+        <number>0</number>
+       </property>
+       <property name="rightMargin">
+        <number>0</number>
+       </property>
+       <property name="bottomMargin">
+        <number>0</number>
+       </property>
+       <property name="spacing">
         <number>6</number>
        </property>
-       <item row="1" column="0" >
-        <widget class="QLabel" name="label_3" >
-         <property name="text" >
+       <item row="1" column="0">
+        <widget class="QLabel" name="label_3">
+         <property name="text">
           <string>Destination</string>
          </property>
         </widget>
        </item>
-       <item row="0" column="0" >
-        <widget class="QLabel" name="label" >
-         <property name="text" >
+       <item row="0" column="0">
+        <widget class="QLabel" name="label">
+         <property name="text">
           <string>Source</string>
          </property>
         </widget>
        </item>
-       <item row="0" column="2" >
-        <widget class="QLabel" name="sourceFileLabel" >
-         <property name="text" >
+       <item row="0" column="2">
+        <widget class="QLabel" name="sourceFileLabel">
+         <property name="text">
           <string>[sourceFile]</string>
          </property>
         </widget>
        </item>
-       <item row="1" column="2" >
-        <widget class="QLabel" name="destinationFileLabel" >
-         <property name="text" >
+       <item row="1" column="2">
+        <widget class="QLabel" name="destinationFileLabel">
+         <property name="text">
           <string>[destinationFile]</string>
          </property>
         </widget>
        </item>
-       <item row="1" column="1" >
-        <widget class="QLineEdit" name="destLineEdit" >
-         <property name="sizePolicy" >
-          <sizepolicy>
-           <hsizetype>7</hsizetype>
-           <vsizetype>0</vsizetype>
+       <item row="1" column="1">
+        <widget class="QLineEdit" name="destLineEdit">
+         <property name="sizePolicy">
+          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
-         <property name="focusPolicy" >
+         <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
-         <property name="text" >
+         <property name="text">
           <string/>
          </property>
-         <property name="readOnly" >
+         <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
-       <item row="0" column="1" >
-        <widget class="QLineEdit" name="sourceLineEdit" >
-         <property name="sizePolicy" >
-          <sizepolicy>
-           <hsizetype>7</hsizetype>
-           <vsizetype>0</vsizetype>
+       <item row="0" column="1">
+        <widget class="QLineEdit" name="sourceLineEdit">
+         <property name="sizePolicy">
+          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
-         <property name="focusPolicy" >
+         <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
-         <property name="readOnly" >
+         <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
@@ -117,29 +138,38 @@
     </layout>
    </item>
    <item>
-    <widget class="QLabel" name="label_2" >
-     <property name="text" >
+    <widget class="QLabel" name="label_2">
+     <property name="text">
       <string>Destination already exists.</string>
      </property>
-     <property name="alignment" >
+     <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
-    <layout class="QHBoxLayout" >
-     <property name="margin" >
+    <layout class="QHBoxLayout">
+     <property name="spacing">
+      <number>6</number>
+     </property>
+     <property name="leftMargin">
       <number>0</number>
      </property>
-     <property name="spacing" >
-      <number>6</number>
+     <property name="topMargin">
+      <number>0</number>
+     </property>
+     <property name="rightMargin">
+      <number>0</number>
+     </property>
+     <property name="bottomMargin">
+      <number>0</number>
      </property>
      <item>
       <spacer>
-       <property name="orientation" >
+       <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
-       <property name="sizeHint" >
+       <property name="sizeHint" stdset="0">
         <size>
          <width>48</width>
          <height>31</height>
@@ -148,46 +178,46 @@
       </spacer>
      </item>
      <item>
-      <widget class="QPushButton" name="skipButton" >
-       <property name="text" >
+      <widget class="QPushButton" name="skipButton">
+       <property name="text">
         <string>Skip</string>
        </property>
       </widget>
      </item>
      <item>
-      <widget class="QPushButton" name="skipAllButton" >
-       <property name="text" >
+      <widget class="QPushButton" name="skipAllButton">
+       <property name="text">
         <string>Skip All</string>
        </property>
       </widget>
      </item>
      <item>
-      <widget class="QPushButton" name="overwriteButton" >
-       <property name="text" >
+      <widget class="QPushButton" name="overwriteButton">
+       <property name="text">
         <string>Overwrite</string>
        </property>
       </widget>
      </item>
      <item>
-      <widget class="QPushButton" name="overwriteAllButton" >
-       <property name="text" >
+      <widget class="QPushButton" name="overwriteAllButton">
+       <property name="text">
         <string>Overwrite All</string>
        </property>
       </widget>
      </item>
      <item>
-      <widget class="QPushButton" name="cancelButton" >
-       <property name="text" >
+      <widget class="QPushButton" name="cancelButton">
+       <property name="text">
         <string>Cancel</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer>
-       <property name="orientation" >
+       <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
-       <property name="sizeHint" >
+       <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
@@ -198,8 +228,8 @@
     </layout>
    </item>
   </layout>
-  <widget class="QWidget" name="widget" >
-   <property name="geometry" >
+  <widget class="QWidget" name="widget" native="true">
+   <property name="geometry">
     <rect>
      <x>9</x>
      <y>83</y>
@@ -209,7 +239,6 @@
    </property>
   </widget>
  </widget>
- <pixmapfunction></pixmapfunction>
  <resources/>
  <connections/>
 </ui>
