diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtservice-orig/src/qtservice.cpp /home/canita/work/patches/qtservice/src/qtservice.cpp
--- /home/canita/work/patches/qtservice-orig/src/qtservice.cpp	2015-06-29 06:40:52.000000000 +0300
+++ /home/canita/work/patches/qtservice/src/qtservice.cpp	2017-01-28 12:35:57.000000000 +0200
@@ -38,26 +38,41 @@
 **
 ****************************************************************************/
 
-#include "qtservice.h"
-#include "qtservice_p.h"
-#include <QCoreApplication>
+
+#include <cppdevtk/util/qt_service.hpp>
+#include <cppdevtk/base/cassert.hpp>
+#include <cppdevtk/base/unused.hpp>
+#include <cppdevtk/base/logger.hpp>
+#include "qt_service_p.hpp"
+
+#include <QtCore/QtGlobal>
+#include <QtCore/QCoreApplication>
+#include <QtCore/QTimer>
+#include <QtCore/QVector>
+#include <QtCore/QProcess>
+
 #include <stdio.h>
-#include <QTimer>
-#include <QVector>
-#include <QProcess>
 
 #if defined(QTSERVICE_DEBUG)
-#include <QDebug>
-#include <QString>
-#include <QFile>
-#include <QTime>
-#include <QMutex>
+#include <QtCore/QDebug>
+#include <QtCore/QString>
+#include <QtCore/QFile>
+#include <QtCore/QTime>
+#include <QtCore/QMutex>
 #if defined(Q_OS_WIN32)
 #include <qt_windows.h>
 #else
 #include <unistd.h>
 #include <stdlib.h>
 #endif
+#endif
+
+
+namespace cppdevtk {
+namespace util {
+
+
+#if defined(QTSERVICE_DEBUG)
 
 static QFile* f = 0;
 
@@ -73,7 +88,7 @@
     f = 0;
 }
 
-#if QT_VERSION >= 0x050000
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
 void qtServiceLogDebug(QtMsgType type, const QMessageLogContext &context, const QString &msg)
 #else
 void qtServiceLogDebug(QtMsgType type, const char* msg)
@@ -124,11 +139,11 @@
         break;
     }
 
-#if QT_VERSION >= 0x050400
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 4, 0))
     s += qFormatLogMessage(type, context, msg).toLocal8Bit();
-#elif QT_VERSION >= 0x050000
+#elif (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
     s += msg.toLocal8Bit();
-    Q_UNUSED(context)
+    ::cppdevtk::base::SuppressUnusedWarning(context);
 #else
     s += msg;
 #endif
@@ -322,7 +337,16 @@
     arguments << QLatin1String("-i");
     arguments << account;
     arguments << password;
+#   if (!CPPDEVTK_PLATFORM_IOS)
     return (QProcess::execute(serviceFilePath, arguments) == 0);
+#   else
+    // TODO: iOS port
+    ::cppdevtk::base::SuppressUnusedWarning(serviceFilePath);
+    ::cppdevtk::base::SuppressUnusedWarning(account);
+    ::cppdevtk::base::SuppressUnusedWarning(password);
+    CPPDEVTK_COMPILER_WARNING("QtServiceController::install() not ported on iOS");
+    return false;
+#   endif
 }
 
 
@@ -431,7 +455,10 @@
 private:
     QtServiceBasePrivate *d_ptr;
 };
-#include "qtservice.moc"
+
+
+#include "qt_service.moc"
+
 
 QtServiceBase *QtServiceBasePrivate::instance = 0;
 
@@ -639,7 +666,7 @@
 QtServiceBase::QtServiceBase(int argc, char **argv, const QString &name)
 {
 #if defined(QTSERVICE_DEBUG)
-#  if QT_VERSION >= 0x050000
+#  if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
     qInstallMessageHandler(qtServiceLogDebug);
 #  else
     qInstallMsgHandler(qtServiceLogDebug);
@@ -647,16 +674,16 @@
     qAddPostRoutine(qtServiceCloseDebugLog);
 #endif
 
-    Q_ASSERT(!QtServiceBasePrivate::instance);
+    CPPDEVTK_ASSERT(!QtServiceBasePrivate::instance);
     QtServiceBasePrivate::instance = this;
 
     QString nm(name);
     if (nm.length() > 255) {
-	qWarning("QtService: 'name' is longer than 255 characters.");
+	CPPDEVTK_LOG_WARN("'name' is longer than 255 characters.");
 	nm.truncate(255);
     }
     if (nm.contains('\\')) {
-	qWarning("QtService: 'name' contains backslashes '\\'.");
+	CPPDEVTK_LOG_WARN("'name' contains backslashes '\\'.");
 	nm.replace((QChar)'\\', (QChar)'\0');
     }
 
@@ -1127,4 +1154,5 @@
 */
 
 
-
+}	// namespace util
+}	// namespace cppdevtk
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtservice-orig/src/qtservice.h /home/canita/work/patches/qtservice/src/qtservice.h
--- /home/canita/work/patches/qtservice-orig/src/qtservice.h	2015-06-29 06:40:52.000000000 +0300
+++ /home/canita/work/patches/qtservice/src/qtservice.h	2017-01-28 12:35:55.000000000 +0200
@@ -38,31 +38,28 @@
 **
 ****************************************************************************/
 
-#ifndef QTSERVICE_H
-#define QTSERVICE_H
 
-#include <QCoreApplication>
+#ifndef CPPDEVTK_UTIL_QT_SERVICE_HPP_INCLUDED_
+#define CPPDEVTK_UTIL_QT_SERVICE_HPP_INCLUDED_
+
+
+#include "config.hpp"
+#include <cppdevtk/base/unused.hpp>
+
+#include <QtCore/QCoreApplication>
 
-#if defined(Q_OS_WIN)
-#  if !defined(QT_QTSERVICE_EXPORT) && !defined(QT_QTSERVICE_IMPORT)
-#    define QT_QTSERVICE_EXPORT
-#  elif defined(QT_QTSERVICE_IMPORT)
-#    if defined(QT_QTSERVICE_EXPORT)
-#      undef QT_QTSERVICE_EXPORT
-#    endif
-#    define QT_QTSERVICE_EXPORT __declspec(dllimport)
-#  elif defined(QT_QTSERVICE_EXPORT)
-#    undef QT_QTSERVICE_EXPORT
-#    define QT_QTSERVICE_EXPORT __declspec(dllexport)
-#  endif
-#else
-#  define QT_QTSERVICE_EXPORT
-#endif
 
 class QStringList;
+
+
+namespace cppdevtk {
+namespace util {
+
+
 class QtServiceControllerPrivate;
 
-class QT_QTSERVICE_EXPORT QtServiceController
+
+class CPPDEVTK_UTIL_API QtServiceController
 {
     Q_DECLARE_PRIVATE(QtServiceController)
 public:
@@ -97,9 +94,11 @@
     QtServiceControllerPrivate *d_ptr;
 };
 
+
 class QtServiceBasePrivate;
 
-class QT_QTSERVICE_EXPORT QtServiceBase
+
+class CPPDEVTK_UTIL_API QtServiceBase
 {
     Q_DECLARE_PRIVATE(QtServiceBase)
 public:
@@ -158,6 +157,7 @@
     QtServiceBasePrivate *d_ptr;
 };
 
+
 template <typename Application>
 class QtService : public QtServiceBase
 {
@@ -177,7 +177,7 @@
     {
         app = new Application(argc, argv);
         QCoreApplication *a = app;
-        Q_UNUSED(a);
+        ::cppdevtk::base::SuppressUnusedWarning(a);
     }
 
     virtual int executeApplication()
@@ -187,6 +187,12 @@
     Application *app;
 };
 
+
 Q_DECLARE_OPERATORS_FOR_FLAGS(QtServiceBase::ServiceFlags)
 
-#endif // QTSERVICE_H
+
+}	// namespace util
+}	// namespace cppdevtk
+
+
+#endif	// CPPDEVTK_UTIL_QT_SERVICE_HPP_INCLUDED_
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtservice-orig/src/qtservice_p.h /home/canita/work/patches/qtservice/src/qtservice_p.h
--- /home/canita/work/patches/qtservice-orig/src/qtservice_p.h	2015-06-29 06:40:52.000000000 +0300
+++ /home/canita/work/patches/qtservice/src/qtservice_p.h	2017-01-28 12:35:57.000000000 +0200
@@ -38,11 +38,20 @@
 **
 ****************************************************************************/
 
-#ifndef QTSERVICE_P_H
-#define QTSERVICE_P_H
 
-#include <QStringList>
-#include "qtservice.h"
+#ifndef CPPDEVTK_UTIL_QT_SERVICE_P_HPP_INCLUDED_
+#define CPPDEVTK_UTIL_QT_SERVICE_P_HPP_INCLUDED_
+
+
+#include <cppdevtk/util/config.hpp>
+#include <cppdevtk/util/qt_service.hpp>
+
+#include <QtCore/QStringList>
+
+
+namespace cppdevtk {
+namespace util {
+
 
 class QtServiceControllerPrivate
 {
@@ -84,4 +93,9 @@
     class QtServiceSysPrivate *sysd;
 };
 
-#endif
+
+}	// namespace util
+}	// namespace cppdevtk
+
+
+#endif	// CPPDEVTK_UTIL_QT_SERVICE_P_HPP_INCLUDED_
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtservice-orig/src/qtservice_unix.cpp /home/canita/work/patches/qtservice/src/qtservice_unix.cpp
--- /home/canita/work/patches/qtservice-orig/src/qtservice_unix.cpp	2015-06-29 06:40:52.000000000 +0300
+++ /home/canita/work/patches/qtservice/src/qtservice_unix.cpp	2017-01-28 12:35:57.000000000 +0200
@@ -38,15 +38,26 @@
 **
 ****************************************************************************/
 
-#include "qtservice.h"
-#include "qtservice_p.h"
-#include "qtunixsocket.h"
-#include "qtunixserversocket.h"
-#include <QCoreApplication>
-#include <QStringList>
-#include <QFile>
-#include <QTimer>
-#include <QDir>
+#include <cppdevtk/util/qt_service.hpp>
+#if (!CPPDEVTK_PLATFORM_UNIX)
+#	error "This file is Unix specific!!!"
+#endif
+
+#include <cppdevtk/base/unused.hpp>
+#include "qt_service_p.hpp"
+#include "qt_unix_socket_unx.hpp"
+#include "qt_unix_server_socket_unx.hpp"
+
+#include <QtCore/QtGlobal>
+#include <QtCore/QCoreApplication>
+#include <QtCore/QStringList>
+#include <QtCore/QFile>
+#include <QtCore/QTimer>
+#include <QtCore/QDir>
+#include <QtCore/QMap>
+#include <QtCore/QSettings>
+#include <QtCore/QProcess>
+
 #include <pwd.h>
 #include <fcntl.h>
 #include <unistd.h>
@@ -55,9 +66,11 @@
 #include <syslog.h>
 #include <signal.h>
 #include <sys/stat.h>
-#include <QMap>
-#include <QSettings>
-#include <QProcess>
+
+
+namespace cppdevtk {
+namespace util {
+
 
 static QString encodeName(const QString &name, bool allowUpper = false)
 {
@@ -215,7 +228,14 @@
         return false;
     if (isRunning())
         return false;
+#   if (!CPPDEVTK_PLATFORM_IOS)
     return QProcess::startDetached(serviceFilePath(), arguments);
+#   else
+    // TODO: iOS port
+    ::cppdevtk::base::SuppressUnusedWarning(arguments);
+    CPPDEVTK_COMPILER_WARNING("QtServiceController::start() not ported on iOS");
+    return false;
+#   endif
 }
 
 bool QtServiceController::stop()
@@ -281,7 +301,7 @@
     QtServiceBase::ServiceFlags serviceFlags;
 
 protected:
-#if QT_VERSION >= 0x050000
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
     void incomingConnection(qintptr socketDescriptor);
 #else
     void incomingConnection(int socketDescriptor);
@@ -307,7 +327,7 @@
 	delete[] ident;
 }
 
-#if QT_VERSION >= 0x050000
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
 void QtServiceSysPrivate::incomingConnection(qintptr socketDescriptor)
 #else
 void QtServiceSysPrivate::incomingConnection(int socketDescriptor)
@@ -377,7 +397,9 @@
     return "";
 }
 
-#include "qtservice_unix.moc"
+
+#include "qt_service_unx.moc"
+
 
 bool QtServiceBasePrivate::sysInit()
 {
@@ -413,13 +435,19 @@
     // Could just call controller.start() here, but that would fail if
     // we're not installed. We do not want to strictly require installation.
     ::setenv("QTSERVICE_RUN", "1", 1);  // Tell the detached process it's it
+#   if (!CPPDEVTK_PLATFORM_IOS)
     return QProcess::startDetached(filePath(), args.mid(1), "/");
+#   else
+    // TODO: iOS port
+    CPPDEVTK_COMPILER_WARNING("QtServiceBasePrivate::start() not ported on iOS");
+    return false;
+#   endif
 }
 
 bool QtServiceBasePrivate::install(const QString &account, const QString &password)
 {
-    Q_UNUSED(account)
-    Q_UNUSED(password)
+    ::cppdevtk::base::SuppressUnusedWarning(account);
+    ::cppdevtk::base::SuppressUnusedWarning(password);
     QSettings settings(QSettings::SystemScope, "QtSoftware");
 
     settings.beginGroup("services");
@@ -480,3 +508,6 @@
         d_ptr->sysd->serviceFlags = flags;
 }
 
+
+}	// namespace util
+}	// namespace cppdevtk
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtservice-orig/src/qtservice_win.cpp /home/canita/work/patches/qtservice/src/qtservice_win.cpp
--- /home/canita/work/patches/qtservice-orig/src/qtservice_win.cpp	2015-06-29 06:40:52.000000000 +0300
+++ /home/canita/work/patches/qtservice/src/qtservice_win.cpp	2017-01-28 12:35:57.000000000 +0200
@@ -38,30 +38,42 @@
 **
 ****************************************************************************/
 
-#include "qtservice.h"
-#include "qtservice_p.h"
-#include <QCoreApplication>
-#include <QDateTime>
-#include <QFile>
-#include <QLibrary>
-#include <QMutex>
-#include <QSemaphore>
-#include <QProcess>
-#include <QSettings>
-#include <QTextStream>
+
+#include <cppdevtk/util/qt_service.hpp>
+#if (!CPPDEVTK_PLATFORM_WINDOWS)
+#	error "This file is Windows specific!!!"
+#endif
+
+#if (CPPDEVTK_DISABLE_CPPDEVTK_WARNINGS)
+#pragma warning(disable: 4191)	// C4191: 'operator/operation' : unsafe conversion from 'type of expression' to 'type required'
+#endif
+
+#include "qt_service_p.hpp"
+
+#include <QtCore/QCoreApplication>
+#include <QtCore/QDateTime>
+#include <QtCore/QFile>
+#include <QtCore/QLibrary>
+#include <QtCore/QMutex>
+#include <QtCore/QSemaphore>
+#include <QtCore/QProcess>
+#include <QtCore/QSettings>
+#include <QtCore/QTextStream>
 #include <qt_windows.h>
-#include <QWaitCondition>
-#include <QAbstractEventDispatcher>
-#include <QVector>
-#include <QThread>
-#if QT_VERSION >= 0x050000
+#include <QtCore/QWaitCondition>
+#include <QtCore/QAbstractEventDispatcher>
+#include <QtCore/QVector>
+#include <QtCore/QThread>
+#include <QtCore/QtGlobal>
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
 #  include <QAbstractNativeEventFilter>
 #endif
 #include <stdio.h>
 #if defined(QTSERVICE_DEBUG)
-#include <QDebug>
+#include <QtCore/QDebug>
 #endif
 
+
 typedef SERVICE_STATUS_HANDLE(WINAPI*PRegisterServiceCtrlHandler)(const wchar_t*,LPHANDLER_FUNCTION);
 static PRegisterServiceCtrlHandler pRegisterServiceCtrlHandler = 0;
 typedef BOOL(WINAPI*PSetServiceStatus)(SERVICE_STATUS_HANDLE,LPSERVICE_STATUS);
@@ -104,6 +116,11 @@
 #define RESOLVEA(name) p##name = (P##name)lib.resolve(#name"A");
 #define RESOLVEW(name) p##name = (P##name)lib.resolve(#name"W");
 
+
+namespace cppdevtk {
+namespace util {
+
+
 static bool winServiceInit()
 {
     if (!pOpenSCManager) {
@@ -418,7 +435,7 @@
 }
 
 #if defined(QTSERVICE_DEBUG)
-#  if QT_VERSION >= 0x050000
+#  if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
 extern void qtServiceLogDebug(QtMsgType type, const QMessageLogContext &context, const QString &msg);
 #  else
 extern void qtServiceLogDebug(QtMsgType type, const char* msg);
@@ -437,7 +454,7 @@
     case Information: //fall through
     default: dbgMsg += "Information] "; break;
     }
-#  if QT_VERSION >= 0x050000
+#  if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
     qtServiceLogDebug((QtMsgType)-1, QMessageLogContext(), QLatin1String(dbgMsg) + message);
 #  else
     qtServiceLogDebug((QtMsgType)-1, (dbgMsg + message.toAscii()).constData());
@@ -458,7 +475,7 @@
     if (h) {
         const wchar_t *msg = (wchar_t*)message.utf16();
         const char *bindata = data.size() ? data.constData() : 0;
-        pReportEvent(h, wType, category, id, 0, 1, data.size(),(const wchar_t **)&msg,
+        pReportEvent(h, wType, (WORD)category, id, 0, 1, data.size(),(const wchar_t **)&msg,
                      const_cast<char *>(bindata));
         pDeregisterEventSource(h);
     }
@@ -497,7 +514,7 @@
     QStringList serviceArgs;
 
     static QtServiceSysPrivate *instance;
-#if QT_VERSION < 0x050000
+#if (QT_VERSION < QT_VERSION_CHECK(5, 0, 0))
     static QCoreApplication::EventFilter nextFilter;
 #endif
 
@@ -524,7 +541,7 @@
 
 
 QtServiceSysPrivate *QtServiceSysPrivate::instance = 0;
-#if QT_VERSION < 0x050000
+#if (QT_VERSION < QT_VERSION_CHECK(5, 0, 0))
 QCoreApplication::EventFilter QtServiceSysPrivate::nextFilter = 0;
 #endif
 
@@ -688,7 +705,8 @@
     return control;
 }
 
-#include "qtservice_win.moc"
+
+#include "qt_service_win.moc"
 
 
 class HandlerThread : public QThread
@@ -731,7 +749,7 @@
   Ignore WM_ENDSESSION system events, since they make the Qt kernel quit
 */
 
-#if QT_VERSION >= 0x050000
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
 
 class QtServiceAppEventFilter : public QAbstractNativeEventFilter
 {
@@ -828,7 +846,7 @@
     if (!app)
         return false;
 
-#if QT_VERSION >= 0x050000
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
     QAbstractEventDispatcher::instance()->installNativeEventFilter(qtServiceAppEventFilter());
 #else
     QtServiceSysPrivate::nextFilter = app->setEventFilter(myEventFilter);
@@ -950,3 +968,5 @@
 }
 
 
+}	// namespace util
+}	// namespace cppdevtk
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtservice-orig/src/qtunixserversocket.cpp /home/canita/work/patches/qtservice/src/qtunixserversocket.cpp
--- /home/canita/work/patches/qtservice-orig/src/qtunixserversocket.cpp	2015-06-29 06:40:52.000000000 +0300
+++ /home/canita/work/patches/qtservice/src/qtunixserversocket.cpp	2017-01-28 12:35:57.000000000 +0200
@@ -38,18 +38,29 @@
 **
 ****************************************************************************/
 
-#include "qtunixserversocket.h"
+
+#include "qt_unix_server_socket_unx.hpp"
+#if (!CPPDEVTK_PLATFORM_UNIX)
+#	error "This file is Unix specific!!!"
+#endif
+
 #include <sys/types.h>
 #include <sys/un.h>
 #include <sys/socket.h>
 #include <string.h>
 #include <unistd.h>
 
+
 #ifndef SUN_LEN
 #define SUN_LEN(ptr) ((size_t)(((struct sockaddr_un *) 0)->sun_path) \
                       +strlen ((ptr)->sun_path))
 #endif
 
+
+namespace cppdevtk {
+namespace util {
+
+
 QtUnixServerSocket::QtUnixServerSocket(const QString &path, QObject *parent)
     : QTcpServer(parent)
 {
@@ -90,3 +101,7 @@
         path_.clear();
     }
 }
+
+
+}	// namespace util
+}	// namespace cppdevtk
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtservice-orig/src/qtunixserversocket.h /home/canita/work/patches/qtservice/src/qtunixserversocket.h
--- /home/canita/work/patches/qtservice-orig/src/qtunixserversocket.h	2015-06-29 06:40:52.000000000 +0300
+++ /home/canita/work/patches/qtservice/src/qtunixserversocket.h	2017-01-28 12:35:57.000000000 +0200
@@ -38,10 +38,22 @@
 **
 ****************************************************************************/
 
-#ifndef QTUNIXSERVERSOCKET_H
-#define QTUNIXSERVERSOCKET_H
 
-#include <QTcpServer>
+#ifndef CPPDEVTK_UTIL_QT_UNIX_SERVER_SOCKET_UNX_HPP_INCLUDED_
+#define CPPDEVTK_UTIL_QT_UNIX_SERVER_SOCKET_UNX_HPP_INCLUDED_
+
+
+#include <cppdevtk/util/config.hpp>
+#if (!CPPDEVTK_PLATFORM_UNIX)
+#	error "This file is Unix specific!!!"
+#endif
+
+#include <QtNetwork/QTcpServer>
+
+
+namespace cppdevtk {
+namespace util {
+
 
 class QtUnixServerSocket : public QTcpServer
 {
@@ -58,4 +70,8 @@
 };
 
 
-#endif
+}	// namespace util
+}	// namespace cppdevtk
+
+
+#endif	// CPPDEVTK_UTIL_QT_UNIX_SERVER_SOCKET_UNX_HPP_INCLUDED_
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtservice-orig/src/qtunixsocket.cpp /home/canita/work/patches/qtservice/src/qtunixsocket.cpp
--- /home/canita/work/patches/qtservice-orig/src/qtunixsocket.cpp	2015-06-29 06:40:52.000000000 +0300
+++ /home/canita/work/patches/qtservice/src/qtunixsocket.cpp	2017-01-28 12:35:57.000000000 +0200
@@ -38,18 +38,29 @@
 **
 ****************************************************************************/
 
-#include "qtunixsocket.h"
+
+#include "qt_unix_socket_unx.hpp"
+#if (!CPPDEVTK_PLATFORM_UNIX)
+#	error "This file is Unix specific!!!"
+#endif
+
 #include <sys/types.h>
 #include <sys/un.h>
 #include <sys/socket.h>
 #include <string.h>
 #include <unistd.h>
 
+
 #ifndef SUN_LEN
 #define SUN_LEN(ptr) ((size_t)(((struct sockaddr_un *) 0)->sun_path) \
                       +strlen ((ptr)->sun_path))
 #endif
 
+
+namespace cppdevtk {
+namespace util {
+
+
 QtUnixSocket::QtUnixSocket(QObject *parent)
     : QTcpSocket(parent)
 {
@@ -76,3 +87,7 @@
     }
     return ret;
 }
+
+
+}	// namespace util
+}	// namespace cppdevtk
diff -U 3 -H -d -r -N -- /home/canita/work/patches/qtservice-orig/src/qtunixsocket.h /home/canita/work/patches/qtservice/src/qtunixsocket.h
--- /home/canita/work/patches/qtservice-orig/src/qtunixsocket.h	2015-06-29 06:40:52.000000000 +0300
+++ /home/canita/work/patches/qtservice/src/qtunixsocket.h	2017-01-28 12:35:57.000000000 +0200
@@ -38,10 +38,22 @@
 **
 ****************************************************************************/
 
-#ifndef QTUNIXSOCKET_H
-#define QTUNIXSOCKET_H
 
-#include <QTcpSocket>
+#ifndef CPPDEVTK_UTIL_QT_UNIX_SOCKET_UNX_HPP_INCLUDED_
+#define CPPDEVTK_UTIL_QT_UNIX_SOCKET_UNX_HPP_INCLUDED_
+
+
+#include <cppdevtk/util/config.hpp>
+#if (!CPPDEVTK_PLATFORM_UNIX)
+#	error "This file is Unix specific!!!"
+#endif
+
+#include <QtNetwork/QTcpSocket>
+
+
+namespace cppdevtk {
+namespace util {
+
 
 class QtUnixSocket : public QTcpSocket
 {
@@ -52,4 +64,9 @@
     bool connectTo(const QString &path);
 };
 
-#endif
+
+}	// namespace util
+}	// namespace cppdevtk
+
+
+#endif	// CPPDEVTK_UTIL_QT_UNIX_SOCKET_UNX_HPP_INCLUDED_
